lumicks/pylake/detail/image.py

```diff
@@ -147,15 +147,28 @@ def round_up(size, n):
         return int(math.ceil(size / n)) * n
 
     data = data[valid_idx]
-    data.resize(round_up(data.size, pixel_size))
```

Comment:
Maybe get a win-win situation here and avoid the `try` block and while even increasing performance. Something along the lines of:
```python
data[:infowave.size] = data[valid_idx]
data = data[:round_up(data.size, pixel_size)]
```
So we only shrink the data once instead of shrinking it and growing it. This would also avoid `.resize()` and the debugger reference issues. The above is just a quick idea I had -- there could be some major flaw that I haven't considered.

Reply:
I like it, but there's a potential risk here. This means that we truly modify data in-place, which means that the outside behaviour changes (as we are truly modifying the incoming array). Before the initial `data = data[valid_idx]` made a copy.

It's a bit of a concern, since it could cause some difficult to track down bugs. If we do it, we should document it very clearly. I did a quick scan of all the other functions using reconstruct_image, and for now, it seems that the behaviours wouldn't change as the data for this function typically comes from _get_photon_count (both for timestamps and counts), which just makes a Continuous from the h5, which always creates a new numpy array. However, if we were to cache this in the future, we'd get in trouble though.

Maybe slightly less sexy, but:
```
    resized_data = np.zeros(round_up(infowave.size, pixel_size))
    resized_data[:infowave.size] = data[valid_idx]
    pixels = reduce(resized_data.reshape(-1, pixel_size), axis=1)
```

Would still make a copy of the input.

I don't know, what do you think? Speed or safety?

Another thing, but minor, data.size should be infowave.size in the round_up.

Reply:
No, definitely don't modify the input data. I like your suggestion. After that, the other two `.resize()` calls might be omitted the same way.

Reply:
It felt a bit like a C-api, modifying the input ☠️ 

Alright, I changed it to just pre-allocate now.